storey/steps/sample.py

```diff
@@ -0,0 +1,67 @@
+from datetime import datetime, timezone
+from typing import Optional, Union
+
+from storey import Filter, Event
+
+
+class Sample(Filter):
+    """Samples stream events, passes a single event downstream after either reaching 'rate_count' or 'rate_seconds'"""
+
+    def __init__(
+        self,
+        rate_count: Optional[int] = None,
+        rate_seconds: Optional[Union[int, float]] = None,
+        **kwargs,
+    ):
+        super().__init__(lambda event: self._sample(event), **kwargs)
+
+        if rate_count is None and rate_seconds is None:
+            raise ValueError(
+                f"Either rate_count or rate_seconds (or both) must be initialized"
+            )
+
+        if rate_count is not None and rate_count <= 0:
+            raise ValueError(f"Expected rate_count > 0, got {rate_count}")
+
+        if rate_seconds is not None and rate_seconds <= 0:
+            raise ValueError(f"Expected rate_seconds > 0, got {rate_seconds}")
+
+        self._rate_count = rate_count
+        self._rate_seconds = rate_seconds
+
+        self._count = None
+        self._seconds = None
```

Comment:
rename to more meaningful names, something like `event_count` and `last_timestamp`